includes/Core/Util/WP_Context_Switcher_Trait.php

```diff
@@ -47,22 +48,35 @@ protected static function with_frontend_context() {
 	 * @since n.e.x.t
 	 *
 	 * @param string $screen_id WordPress screen ID.
-	 * @return callable Closure that restores context and returns true.
 	 */
 	private static function switch_current_screen( $screen_id ) {
 		global $current_screen;
 
 		require_once ABSPATH . 'wp-admin/includes/class-wp-screen.php';
 		require_once ABSPATH . 'wp-admin/includes/screen.php';
 
+		$current_screen = \WP_Screen::get( $screen_id ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+	}
+
+	/**
+	 * Returns the closure to restore the current screen.
+	 *
+	 * Calling the closure will restore the `$current_screen` global to what it was set to at the time of calling
+	 * this method.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Closure that restores context.
+	 */
+	private static function get_restore_closure() {
```

Comment:
This name is a bit too generic to use in a trait I think; even though it's private, it could still have a naming collision. How about something like this instead?
```suggestion
	private static function get_restore_current_screen_closure() {
```